### PR TITLE
Creating wrapper Object for primitive data Types

### PR DESCRIPTION
**Pull request by RoshniNaveenaS**
Fix for Issue - https://github.tools.sap/cap/issues/issues/15789

### CODE REVIEW

File: lib/compile/csdl2openapi.js [hunk 1]
```diff
@@ -2271,6 +2271,16 @@ see [Expand](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-prot
             s['x-sap-odm-oid-reference-entity-name'] = element['@ODM.oidReference'].entityName
         }
 
+        //for primitive types creating a wrapper object
+        if (element.$Type?.startsWith('Edm.')) {
```
Comment (jeevitha011) [inline lib/compile/csdl2openapi.js:2275]: Should we not look at the schema return type instead of this `Edm.` check?

File: test/lib/compile/csdl2openapi.test.js [hunk 2]
```diff
@@ -490,28 +490,42 @@ describe('Edge cases', function () {
         };
         const openapi = lib.csdl2openapi(csdl, {});
         assert.deepStrictEqual(
-            openapi.components.schemas["jsonExamples.typeDefinitionOld"],
-            {
-                title: "typeDefinitionOld",
+          openapi.components.schemas["jsonExamples.typeDefinitionOld"],
+          {
+            title: "typeDefinitionOld",
+            type: "object",
```
Comment (jeevitha011) [inline test/lib/compile/csdl2openapi.test.js:496]: why are we adding wrapper object for `type: "object"`?

File: test/lib/compile/csdl2openapi.test.js [hunk 3]
```diff
@@ -490,28 +490,42 @@ describe('Edge cases', function () {
         };
         const openapi = lib.csdl2openapi(csdl, {});
         assert.deepStrictEqual(
-            openapi.components.schemas["jsonExamples.typeDefinitionOld"],
-            {
-                title: "typeDefinitionOld",
+          openapi.components.schemas["jsonExamples.typeDefinitionOld"],
```
Comment (jeevitha011) [inline test/lib/compile/csdl2openapi.test.js:493]: shouldn't the wrapper obj be added in the response objects only?

File: lib/compile/csdl2openapi.js (hunk 1 above)
Comment (RoshniNaveenaS) [reply lib/compile/csdl2openapi.js:2289]: yeah, made the changes.
Checked with Ralf's repo and have added similarly.

File: test/lib/compile/csdl2openapi.test.js (hunk 2 above)
Comment (RoshniNaveenaS) [reply test/lib/compile/csdl2openapi.test.js:496]: corrected

File: test/lib/compile/csdl2openapi.test.js (hunk 3 above)
Comment (RoshniNaveenaS) [reply test/lib/compile/csdl2openapi.test.js:493]: corrected

**Review by jeevitha011 — APPROVED** [review]
